erlang: use nixpkgs zlib, fix darwin

adamcstephens opened the pull request:
Darwin is failing with the bundled zlib. We should be using the nixpkgs zlib anyway, so add to buildInputs for all.

Removed legacy Darwin sdk pattern.

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

adamcstephens committed:
erlang: use nixpkgs zlib, fix darwin

Darwin is failing with the bundled zlib. We should be using the nixpkgs
zlib anyway, so add to buildInputs for all.

Removed legacy Darwin sdk pattern.

lucasewd commented:
# [nixpkgs-reviewd](https://github.com/nixosbrasil/nixpkgs-reviewd) workflow started

**Status and logs**: https://github.com/nixosbrasil/nixpkgs-reviewd/actions/runs/12485284153

happysalada commented:
Did you forget to push the darwin update to newer sdk ?

adamcstephens commented:
stdenv on Darwin includes a default SDK now.

> Darwin needs an SDK to build software. The SDK provides a default set of frameworks and libraries to build software, most of which are specific to Darwin. There are multiple versions of the SDK packages in Nixpkgs, but one is included by default in the stdenv. Usually, you don’t have to change or pick a different SDK. When in doubt, use the default.

https://github.com/NixOS/nixpkgs/blob/master/doc/stdenv/platform-notes.chapter.md

adamcstephens commented:
Fixed my PR description. I removed the old frameworks as they were doing nothing according to https://github.com/NixOS/nixpkgs/blob/master/doc/stdenv/platform-notes.chapter.md#dealing-with-darwinapple_sdkframeworks-sec-darwin-legacy-frameworks

lucasewd commented:
# [nixpkgs-reviewd](https://github.com/nixosbrasil/nixpkgs-reviewd) workflow finished

**Status and logs**: https://github.com/nixosbrasil/nixpkgs-reviewd/actions/runs/12485284153


## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 367981`

---
### `x86_64-darwin`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>couchdb3</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 36 packages built:</summary>
  <ul>
    <li>akkoma</li>
    <li>elixir (elixir_1_17)</li>
    <li>elixir-ls</li>
    <li>elixir_1_10</li>
    <li>elixir_1_11</li>
    <li>elixir_1_12</li>
    <li>elixir_1_13</li>
    <li>elixir_1_14</li>
    <li>elixir_1_15</li>
    <li>elixir_1_16</li>
    <li>elvis-erlang</li>
    <li>erlang (erlang_25)</li>
    <li>erlang-ls</li>
    <li>erlang_24</li>
    <li>erlang_26</li>
    <li>erlang_27</li>
    <li>erlang_nox</li>
    <li>erlfmt</li>
    <li>ex_doc</li>
    <li>gleam</li>
    <li>lexical</li>
    <li>lfe (lfe_2_1)</li>
    <li>livebook</li>
    <li>mercury</li>
    <li>mix2nix</li>
    <li>mobilizon</li>
    <li>next-ls</li>
    <li>plausible</li>
    <li>pleroma</li>
    <li>protoc-gen-elixir</li>
    <li>rabbitmq-server</li>
    <li>rabbitmq-server.doc</li>
    <li>rabbitmq-server.man</li>
    <li>rebar</li>
    <li>rebar3</li>
    <li>tsung</li>
  </ul>
</details>

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 367981`

---
### `aarch64-darwin`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>couchdb3</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 36 packages built:</summary>
  <ul>
    <li>akkoma</li>
    <li>elixir (elixir_1_17)</li>
    <li>elixir-ls</li>
    <li>elixir_1_10</li>
    <li>elixir_1_11</li>
    <li>elixir_1_12</li>
    <li>elixir_1_13</li>
    <li>elixir_1_14</li>
    <li>elixir_1_15</li>
    <li>elixir_1_16</li>
    <li>elvis-erlang</li>
    <li>erlang (erlang_25)</li>
    <li>erlang-ls</li>
    <li>erlang_24</li>
    <li>erlang_26</li>
    <li>erlang_27</li>
    <li>erlang_nox</li>
    <li>erlfmt</li>
    <li>ex_doc</li>
    <li>gleam</li>
    <li>lexical</li>
    <li>lfe (lfe_2_1)</li>
    <li>livebook</li>
    <li>mercury</li>
    <li>mix2nix</li>
    <li>mobilizon</li>
    <li>next-ls</li>
    <li>plausible</li>
    <li>pleroma</li>
    <li>protoc-gen-elixir</li>
    <li>rabbitmq-server</li>
    <li>rabbitmq-server.doc</li>
    <li>rabbitmq-server.man</li>
    <li>rebar</li>
    <li>rebar3</li>
    <li>tsung</li>
  </ul>
</details>

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 367981`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 43 packages built:</summary>
  <ul>
    <li>akkoma</li>
    <li>cl</li>
    <li>couchdb3</li>
    <li>ejabberd</li>
    <li>elixir</li>
    <li>elixir-ls</li>
    <li>elixir_1_10</li>
    <li>elixir_1_11</li>
    <li>elixir_1_12</li>
    <li>elixir_1_13</li>
    <li>elixir_1_14</li>
    <li>elixir_1_15</li>
    <li>elixir_1_16</li>
    <li>elvis-erlang</li>
    <li>erlang</li>
    <li>erlang-ls</li>
    <li>erlang_24</li>
    <li>erlang_26</li>
    <li>erlang_27</li>
    <li>erlang_nox</li>
    <li>erlfmt</li>
    <li>ex_doc</li>
    <li>gleam</li>
    <li>lexical</li>
    <li>lfe</li>
    <li>livebook</li>
    <li>mercury</li>
    <li>mix2nix</li>
    <li>mobilizon</li>
    <li>next-ls</li>
    <li>notmuch-bower</li>
    <li>plausible</li>
    <li>pleroma</li>
    <li>protoc-gen-elixir</li>
    <li>rabbitmq-server</li>
    <li>rabbitmq-server.doc</li>
    <li>rabbitmq-server.man</li>
    <li>rebar</li>
    <li>rebar3</li>
    <li>sonic-pi</li>
    <li>tsung</li>
    <li>wings</li>
    <li>yaws</li>
  </ul>
</details>

adamcstephens commented:
couchdb3 is already failing on darwin on master https://hydra.nixos.org/job/nixpkgs/trunk/couchdb3.aarch64-darwin
